### PR TITLE
apache-arrow-glib 0.9.0 (new formula)

### PR DESCRIPTION
**Pull request by itume**
- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This pull request is Based on https://github.com/Homebrew/homebrew-core/pull/18590
I made some revisions, please check it out.

### CODE REVIEW

**Comment by tschoonj** [comment]
The `gobject-introspection` should be `:build`. This will break the test though... probably better to just check for the presence of the `.typelib` and `.gir` files.

**Comment by tschoonj** [comment]
You could also rewrite the test to compile and run a small test program that uses the C glib bindings (not the GObject Introspection bindings).

For inspiration have a look at e.g. `gtk+3`. Your test should be shorter though.

**Comment by itume** [comment]
I try to fix test referencing gtk+3.rb. thank you for your code review.

**Comment by itume** [comment]
>The gobject-introspection should be :build.
I think run-time dependency is more better. Because typical use of this library is using Apache Arrow via script language, such as Ruby. Would it be acceptable?

**Comment by tschoonj** [comment]
No, over the past two weeks I have removed `gobject-introspection` everywhere as run-time dependency and kept it build-time only. Mainly to speed up building and testing on Jenkins, and because not everybody needs it anyway. This way the gir files get built, but if the user needs to use them, it is up to him/her to install the appropriate gir bindings for the language from where they need to be used. (e.g. `pygobject3`). Note that even the gir bindings won't always depend on `gobject-introspection`

File: Formula/apache-arrow-glib.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <arrow-glib/arrow-glib.h>
+      int main(void){
+        GArrowArray *array;
```
Comment (tschoonj) [inline Formula/apache-arrow-glib.rb]: Please put in a call to a method from the library. We need this to test for linker issues.

For example:

```c
GArrowNullArray *array = garrow_null_array_new(10);
```

File: Formula/apache-arrow-glib.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  homepage "https://arrow.apache.org/"
+  url "https://www.apache.org/dyn/closer.cgi?path=arrow/arrow-0.8.0/apache-arrow-0.8.0.tar.gz"
+  sha256 "b142b883c7107485a296655e8d7421d11891acbca15862f37a4a8c1b13d4e9c4"
+  head "https://github.com/apache/arrow.git"
+
+  depends_on "pkg-config" => :build
+  depends_on "apache-arrow"
+  depends_on "gettext"
+  depends_on "glib"
+  depends_on "gobject-introspection"
```
Comment (tschoonj) [inline Formula/apache-arrow-glib.rb]: As mentioned before, please make `gobject-introspection` `:build`

**Review by tschoonj — CHANGES_REQUESTED** [review]

**Review by tschoonj — APPROVED** [review]

**Comment by tschoonj** [comment]
Thanks for your changes. Looks ready to merge to me

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @itume! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**